clients/privacy-center/pages/consent.tsx

```diff
@@ -11,179 +11,260 @@ import {
+      setConsentItems(updatedConsentItems);
+      setConsentCookie(makeCookieKeyConsent(updatedConsentItems));
+    },
+    [consentOptions]
+  );
 
-      const requestOptions: RequestInit = {
-        method: privacyCenterConfig.identity_verification_required
-          ? "POST"
-          : "GET",
-        headers,
-      };
-      if (privacyCenterConfig.identity_verification_required) {
-        requestOptions.body = JSON.stringify({ code: verificationCode });
-      }
+  /**
+   * Set the consent value for an option in the `consentItems` array. We're storing a whole array in
+   * the state, so we need to use `produce` to modify a single property but still get a new object
+   * that works with React rendering.
+   */
+  const setConsentValue = useCallback(
+    (item: ConsentItem, value: boolean) => {
+      const updatedConsentItems = produce(consentItems, (draftItems) => {
+        const itemToUpdate = draftItems.find(
+          (candidate) => candidate.fidesDataUseKey === item.fidesDataUseKey
+        );
+        if (!itemToUpdate) {
+          return;
+        }
+        itemToUpdate.consentValue = value;
+      });
+      setConsentItems(updatedConsentItems);
+    },
+    [consentItems]
+  );
 
-      const response = await fetch(`${hostUrl}/${verifyUrl}`, requestOptions);
-      const data = (await response.json()) as ApiUserConsents;
-      if (!response.ok) {
-        toast({
-          title: "An error occurred while retrieving user consent preferences",
-          description: (data as any).detail,
-          ...ErrorToastOptions,
-        });
+  /**
+   * When the Id verification method is known, trigger the request that will
+   * return the consent choices saved on the backend.
+   */
+  useEffect(() => {
+    if (!consentRequestId) {
+      toastError({ title: "No consent request in progress." });
+      redirectToIndex();
+      return;
+    }
 
-        router.push("/");
-        return;
-      }
+    if (getIdVerificationConfigQueryResult.isError) {
+      toastError({ error: getIdVerificationConfigQueryResult.error });
+      return;
+    }
 
-      const updatedConsentItems = makeConsentItems(data, consentOptions);
-      setConsentItems(updatedConsentItems);
-      setConsentCookie(makeCookieKeyConsent(updatedConsentItems));
-    };
-    getUserConsents();
-  }, [router, consentRequestId, verificationCode, toast, consentOptions]);
+    if (!getIdVerificationConfigQueryResult.isSuccess) {
+      return;
+    }
 
-  consentItems.forEach((option) => {
-    content.push(
-      <ConsentItemCard
-        key={option.fidesDataUseKey}
-        fidesDataUseKey={option.fidesDataUseKey}
-        name={option.name}
-        description={option.description}
-        highlight={option.highlight}
-        url={option.url}
-        defaultValue={option.defaultValue}
-        consentValue={option.consentValue}
-        setConsentValue={(value) => {
-          /* eslint-disable-next-line no-param-reassign */
```

Comment:
Getting rid of this was the point of the immer stuff in `setConsentValue`. If this state were stored in a slice instead of local state we could just have a reducer do that automatically, but this is a start.